packages/itmat-job-executor/src/jobHandlers/UKB_CSV_UPLOAD_handler.ts

```diff
@@ -12,9 +12,9 @@ export class UKB_CSV_UPLOAD_Handler extends JobHandler {
     // private ukbCurator: UKBCurator;
 
     public async getInstance() {
-        if (!this._instance) {
+        if (!this._instance || !(this._instance instanceof UKB_CSV_UPLOAD_Handler)) {
```

Comment:
Which situation would mean that _instance is not of type `UKB_CSV_UPLOAD_Handler` ? If that's the case is that even safe to replace it with no extra checks ?

Reply:
E.g., when you submit a job of parsing CSV first and then parsing Field/Json, the latter job will still hold the former parser classes, i.e., UKB_CSV_Upload_handler. When checking the 'this', it seems to be bound to the outer class. Thus, once this._instance is firstly instantiated, all following jobs will continue using this instance.

Reply:
@fguitton  @wsy19961129 
is the "this" bound to the dispatcher?
maybe we need to .bind(this) there?

Reply:
So if continuing parsing the same type of file, the 'this._instance' does not need to be changed. It's also possible to return new instance everytime, but as I think it might be more efficient with the current form.

Reply:
> @fguitton @wsy19961129
> is the "this" bound to the dispatcher?
> maybe we need to .bind(this) there?

Yes, bound to the jobDispatcher.